gocov/test.go

```diff
@@ -86,12 +90,24 @@ func runTests(args []string) error {
 		// the JSON coverage output.
 		cmd.Stdout = os.Stderr
 		cmd.Stderr = os.Stderr
-		err := cmd.Run()
+
+		commandChan <- cmd
+	}
+
+	for i := 0; i < runtime.NumCPU(); i++ {
```

Comment:
It should not matter whether -p=x is passed to the package commands; -p just controls how many packages may be built/tested in parallel. It's only interesting when you specify multiple packages.

Running the package tests separately (i.e. one per goroutine) is necessary due to the limitations of the "go" tool -- it will reject attempts to test with -coverprofile if multiple packages are specified.